src/main/scala/com/fulcrumgenomics/fasta/CollectAlternateContigNames.scala

```diff
@@ -192,11 +210,22 @@ class CollectAlternateContigNames
         }
       }
     }
-    // Sort the secondary entries based on descending sequence length.
-    val infos = (primaries ++ secondaries.sortBy(-_.length))
-      .zipWithIndex
-      .map { case (metadata, index) => metadata.copy(index=index) }
-      .toSeq
+
+    // Get the ordering of the molecules
+    val infos: Seq[SequenceMetadata] = existingSequenceDictionary match {
+      case None =>
+        // Sort the secondary entries based on descending sequence length.
+        (primaries ++ secondaries.sortBy(-_.length))
```

Comment:
Why do this sort?  The assembly reports are (in my experience) nicely ordered and this shuffles unplaced/alts/randoms together and by length, instead of keeping them separate and ordered karyotypically.

Reply:
I think this is a hold-over from a genome FASTA that doesn't follow the ordering in the NCBI assembly report.  I suppose the answer is to either manually re-order the input report or manually re-order the output dictionary.